"ldapadd" is just a link to "ldapmodify".  We ship it as "openldapadd"
so detect that as well.

--- openldap-2.6.16/clients/tools/ldapmodify.c.orig
+++ openldap-2.6.16/clients/tools/ldapmodify.c
@@ -233,8 +233,8 @@
 
 	prog = lutil_progname( "ldapmodify", argc, argv );
 
-	/* strncmp instead of strcmp since NT binaries carry .exe extension */
-	ldapadd = ( strncasecmp( prog, "ldapadd", sizeof("ldapadd")-1 ) == 0 );
+	/* detect "openldapadd" as "ldapadd" */
+	ldapadd = ( strcasestr( prog, "ldapadd" ) != NULL );
 
 	tool_init( ldapadd ? TOOL_ADD : TOOL_MODIFY );
 
